### PR TITLE
Feature/9 force login

### PR DESCRIPTION
**Pull request by ovelartelius**
Added logic to force login in environments like integration, preproduction etc.
Used https://ericceric.com/2022/03/04/make-optimizely-dxp-integration-and-preproduction-sites-force-login/ as source

### CODE REVIEW

**Comment by ovelartelius** [comment]
#15

File: Addon.Episerver.EnvironmentSynchronizer/Configuration/EnvironmentSynchronizerOptions.cs [hunk 1]
```diff
@@ -13,7 +12,7 @@ public class EnvironmentSynchronizerOptions
 
         public bool RunInitializationModuleEveryStartup {  get; set;}
 
-        public IList<SiteDefinitionOptions> SiteDefinitions { get; set; }
+		public IList<SiteDefinitionOptions> SiteDefinitions { get; set; }
```
Comment (LinusEkstrom) [inline Addon.Episerver.EnvironmentSynchronizer/Configuration/EnvironmentSynchronizerOptions.cs:15]: Technically this is a breaking change, but since it can be considered an internal API, I think that we can ignore this.

File: Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs [hunk 2]
```diff
@@ -110,5 +125,52 @@ private SiteDefinition GetExistingSiteDefinition(IEnumerable<SiteDefinition> exi
 
             return siteDefinition;
         }
+
+        private void SetForceLogin(SiteDefinition site, EnvironmentSynchronizerSiteDefinition siteDefinitionToUpdate)
```
Comment (LinusEkstrom) [inline Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs]: I suggest renaming this method to RemoveAccessForEveryoneRole

File: Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs [hunk 3]
```diff
@@ -110,5 +125,52 @@ private SiteDefinition GetExistingSiteDefinition(IEnumerable<SiteDefinition> exi
...
+					}
+
+					var foundEveryoneRead = false;
+					var existingEntries = new List<AccessControlEntry>();
+					foreach (var entry in securityDescriptor.Entries)
+					{
+						Logger.Information($"Found AccessControlEntry {entry.Name}-{entry.Access} for site {siteDefinitionToUpdate.Name}.");
+						if (entry.Name == "Everyone")
+						{
+							foundEveryoneRead = true;
```
Comment (LinusEkstrom) [inline Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs]: How do you know that this security descriptor includes read access? If that's true, rename to remove "read" from variable name.

File: README.md [hunk 4]
```diff
@@ -110,14 +117,39 @@ namespace Yoursite.Infrastructure.Environments
...
+```
 In the example above it loads the setting/configuration value from appsettings.json with the name "EnvironmentSettings.Environment".  
 ```json
 {
   "EnvironmentSettings.Environment": "dev",
 }
 ```
-### DXP variable episerver:EnvironmentName support
-If you don´t implement the logic specified above. The DXP variable ´episerver:EnvironmentName´ will be used. More information about the DXP environments and the appsetting can be found on [https://world.episerver.com/documentation/developer-guides/digital-experience-platform/development-considerations/environment-configurations/](https://world.episerver.com/documentation/developer-guides/digital-experience-platform/development-considerations/environment-configurations/)  
+### DXP variable ASPNETCORE_ENVIRONMENT support
```
Comment (LinusEkstrom) [inline README.md:151]: Is this change related or just a fix that should be separated into another PR?

File: Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs (hunk 2 above)
Comment (LinusEkstrom) [inline Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs]: This feels like rather complex business logic that should be unit tested.

**Review by LinusEkstrom — COMMENTED** [review]
In general positive to this change, but gave some comments for improvements.

File: Addon.Episever.EnvironmentSynchronizer.Test/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizerTests.cs [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+using FluentAssertions;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Addon.Episever.EnvironmentSynchronizer.Test.Synchronizers.SiteDefinitions
+{
+	public class SiteDefinitionSynchronizerTests
```
Comment (LinusEkstrom) [inline Addon.Episever.EnvironmentSynchronizer.Test/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizerTests.cs]: Suggestion for easier readability:
1. Rename this test class to RemoveAccessForEveryoneRoleTests. This means that this class is focused on testing that method and you don't need to include that in every test name.
2. Rename the tests so that are written in an easy to read format. You should be able to give the list of names to a project manager, and they should be able to understand. Example:

RemoveAccessForEveryoneRole_NullStartPage_Nothing -> When_everyone_is_not_part_of_accessrights_no_changes_should_be_done

File: README.md (hunk 4 above)
Comment (ovelartelius) [reply README.md:151]: Its a fix. Do you want me to put it in another PR?

File: Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs (hunk 3 above)
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs]: Renamed

File: README.md (hunk 4 above)
Comment (LinusEkstrom) [reply README.md:151]: Yes, please.

File: Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs (hunk 2 above)
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs]: Added unit tests for this logic

Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizer.cs]: Renamed

File: README.md (hunk 4 above)
Comment (ovelartelius) [reply README.md:151]: Can be found in a new PR

File: Addon.Episever.EnvironmentSynchronizer.Test/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizerTests.cs (hunk 5 above)
Comment (ovelartelius) [reply Addon.Episever.EnvironmentSynchronizer.Test/Synchronizers/SiteDefinitions/SiteDefinitionSynchronizerTests.cs]: Renamed class and methods. This will not follow the guide we have for names in Epinova. But that guide is maybe only for internal Epinova.

**Review by LinusEkstrom — APPROVED** [review]
